packages/lib-sourcify/src/lib/CheckedContract.ts

```diff
@@ -748,8 +756,20 @@ function createJsonInputFromMetadata(
  *
  * This will likely moved to server or somewhere else. But keep it here for now.
  */
-export function getIpfsGateway(): string {
-  return process.env.IPFS_GATEWAY || 'https://ipfs.io/ipfs/';
+export function getIpfsGateway(): IpfsGateway {
+  let ipfsGatewaysHeaders;
+  if (process.env.IPFS_GATEWAY_HEADERS) {
+    try {
+      ipfsGatewaysHeaders = JSON.parse(process.env.IPFS_GATEWAY_HEADERS);
+    } catch (error) {
+      logError('Error while parsing IPFS_GATEWAY_HEADERS option', { error });
+      throw new Error('Error while parsing IPFS_GATEWAY_HEADERS option');
+    }
+  }
+  return {
+    url: process.env.IPFS_GATEWAY || 'https://ipfs.io/ipfs/',
```

Comment:
Let's make sure this url has a trailing slash if it doesn't. Later on we append IPFS CID's assuming there's a trailing slash.